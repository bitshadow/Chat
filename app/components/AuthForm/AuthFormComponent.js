// @flow
import React, { Component } from 'react';
import {
  View,
} from 'react-native';

import {
  FormInput,
  FormValidationMessage,
} from 'react-native-elements';
import Form from 'react-native-form';
import Button from '../Button';

type AuthFormProps = {
  onPress: () => void,
  error: string,
  loading: boolean,
  title: string
};

export default class AuthFormComponent extends Component<AuthFormProps, *> {
  static defaultProps = {
    loading: false
  };

  handleSubmit = () => {
    if (this.form) {
      const vals = this.form.getValues();

      this.props.onPress(vals.email, vals.password);
    }
  }

  render() {
    const { title, loading } = this.props;

    return (
      <View>
        <Form ref={e => (this.form = e)}>
          <FormInput
            name="email"
            type="TextInput"
            keyboardType="email-address"
            autoCapitalize="none"
            returnKeyType="next"
            enablesReturnKeyAutomatically
            onSubmitEditing={() => {
              this.ePassword.focus();
            }}
            underlineColorAndroid="transparent"
            placeholder="E-mail Address"
          />
          <FormInput
            name="password"
            type="TextInput"
            ref={e => (this.ePassword = e)}
            autoCapitalize="none"
            returnKeyType="done"
            secureTextEntry
            enablesReturnKeyAutomatically
            selectTextOnFocus
            placeholder="Password"
            onSubmitEditing={this.handleSubmit}
            underlineColorAndroid="transparent"
          />
          <FormValidationMessage
            labelStyle={{
              marginBottom: 10,
              marginRight: 20,
              color: 'red',
              textAlign: 'right',
            }}
          >
            {this.props.error}
          </FormValidationMessage>
        </Form>
        <Button
          title={title}
          type="secondary"
          onPress={this.handleSubmit}
          loading={loading}
        />
      </View>
    );
  }
}
