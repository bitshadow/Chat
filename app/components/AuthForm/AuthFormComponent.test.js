import 'react-native';
import 'isomorphic-fetch';
import React from 'react';
import renderer from 'react-test-renderer';
import AuthFormComponent from './AuthFormComponent';

test('renders correctly', () => {
  const tree = renderer.create(<AuthFormComponent />).toJSON();

  expect(tree).toMatchSnapshot();
});
