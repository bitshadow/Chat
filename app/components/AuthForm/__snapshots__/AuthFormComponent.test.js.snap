// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    customFields={Object {}}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        enablesReturnKeyAutomatically={true}
        keyboardType="email-address"
        name="email"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="E-mail Address"
        returnKeyType="next"
        style={
          Array [
            Object {
              "color": "#86939e",
              "minHeight": 36,
              "width": 750,
            },
            Object {
              "fontSize": 17.5,
            },
            undefined,
          ]
        }
        type="TextInput"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        enablesReturnKeyAutomatically={true}
        name="password"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Password"
        returnKeyType="done"
        secureTextEntry={true}
        selectTextOnFocus={true}
        style={
          Array [
            Object {
              "color": "#86939e",
              "minHeight": 36,
              "width": 750,
            },
            Object {
              "fontSize": 17.5,
            },
            undefined,
          ]
        }
        type="TextInput"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#ff190c",
                "fontSize": 15,
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 5,
              },
              Object {
                "color": "red",
                "marginBottom": 10,
                "marginRight": 20,
                "textAlign": "right",
              },
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {},
        Object {
          "borderRadius": 3,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#745CF9",
            },
            Object {
              "borderRadius": 3,
            },
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "paddingHorizontal": 40,
              "paddingVertical": 14,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                Object {
                  "fontSize": 15,
                },
                undefined,
                Object {
                  "fontWeight": "bold",
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          LOG IN
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginLeft": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="simple-line-icon"
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
